File: HackerRank Problem/Python/Staircase.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the staircase function below.
def staircase(n):
    staircase = []
    for i in range(1,n+1):
        staircase.append((n-i)*" " + (i)*"#")
    sys.stdout.write('\n'.join(staircase))

if __name__ == '__main__':
    n = int(input())
    staircase(n)

File: HackerRank Problem/Ruby/Staircase.rb
#!/bin/ruby

require 'json'
require 'stringio'

# Complete the staircase function below.
def staircase(n)
  spaces=0
  arr = Array.new(n) { |i| Array.new(n) { |j| (i >= j) ? '#' : ' ' } }
  print_matrix(arr.map(&:reverse), spaces)
end

def print_matrix(arr, spaces = 0)
  sep = ' '*(spaces)
  arr.each { |r| puts "#{r.join(sep)}" }
end

n = gets.to_i

staircase n
